src/plugins/data_source_management/public/components/utils.test.ts

```diff
@@ -139,6 +141,51 @@ describe('DataSourceManagement: Utils.ts', () => {
     });
   });
 
+  describe('Test connection - success', () => {
+    let http: jest.Mocked<HttpStart>;
+    const mockSuccess = jest.fn().mockResolvedValue({ body: { success: true } });
+    const mockError = jest.fn().mockRejectedValue(null);
+    beforeEach(() => {
+      http = coreMock.createStart().http;
+      http.post.mockResolvedValue(mockSuccess);
+    });
+    test('Success: Test Connection on create', async () => {
+      await testConnection(http, getDataSourceByIdWithoutCredential.attributes);
+      expect(http.post.mock.calls).toMatchInlineSnapshot(`
+        Array [
+          Array [
+            "/internal/data-source-management/validate",
+            Object {
+              "body": "{\\"id\\":\\"\\",\\"endpoint\\":\\"https://test.com\\",\\"auth\\":{\\"type\\":\\"no_auth\\",\\"credentials\\":null}}",
+            },
+          ],
+        ]
+      `);
+    });
+
+    test('Success: Test Connection on update', async () => {
+      await testConnection(http, getDataSourceByIdWithoutCredential.attributes, 'test1234');
+      expect(http.post.mock.calls).toMatchInlineSnapshot(`
+        Array [
+          Array [
+            "/internal/data-source-management/validate",
+            Object {
+              "body": "{\\"id\\":\\"test1234\\",\\"endpoint\\":\\"https://test.com\\",\\"auth\\":{\\"type\\":\\"no_auth\\",\\"credentials\\":null}}",
+            },
+          ],
+        ]
+      `);
+    });
+    test('failure: Test connection', async () => {
```

Comment:
The names of these tests need rewording to match the general conventions used elsewhere.